clash-lib/src/Clash/Primitives/Intel/ClockGen.hs

```diff
@@ -159,6 +159,8 @@ altpllQsysTemplate bbCtx = pure bbText
   ((_,stripVoid -> kdIn,_):(_,stripVoid -> kdOut,_):_) = bbInputs bbCtx
   KnownDomain _ clkInPeriod _ _ _ _ = kdIn
   KnownDomain _ clkOutPeriod _ _ _ _ = kdOut
+  clkInFreq :: Integer
+  clkInFreq = round (1.0 / (fromInteger @Double clkInPeriod * 1.0e-12))
   clkOutFreq :: Double
   clkOutFreq = (1.0 / (fromInteger clkOutPeriod * 1.0e-12)) / 1e6
```

Comment:
Why do the calculations for `clkInFreq` and `clkOutFreq` differ?

Reply:
Would using `periodToHz` suffice?

Reply:
Originally, I indiscriminately copy-pasted the computation. Then it occurred to me that `clkOutFreq` is in fractional MHz, and I intended integral Hz. By then, I adapted it instead of picking `round . periodToHz`. Anyway, the whole thing was a mistake.